### ISSUE TITLE
[typedef] Ignore parameters of functions that are not a part of a declaration

### ISSUE THREAD

**Issue by vostrnad** [typescript-eslint/typescript-eslint#2285]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-function-return-type": "error",
    "@typescript-eslint/typedef": "error"
  }
}
```

```TS
window.addEventListener('mousemove', (event) => {
  console.log(event);
});
```

**Expected Result**
No lint errors.

**Actual Result**
@typescript-eslint/explicit-function-return-type: *no error*
@typescript-eslint/typedef: Expected event to have a type annotation.

**Additional Info**
Since @typescript-eslint/explicit-function-return-type does not report errors on functions that are only passed as a parameter to another function, I would expect the same from @typescript-eslint/typedef. However it seems that there isn't even a rule option for this.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.6.0` |
| `@typescript-eslint/parser`        | `3.6.0` |
| `TypeScript`                       | `3.9.5` |
| `ESLint`                           | `7.4.0` |
| `node`                             | `12.13.1` |
| `npm`                              | `6.12.1` |

**Comment by bradzacher** [comment]
The rule essentially exists as a stop-gap to help users migrate onto `noImplicitAny`.

We recommend you don't use `typedef`.
The conventions it enforces can slow down your typecheck, and cause your codebase to be overly verbose and sometimes hard to maintain.

As such, we won't be adding more functionality to this rule - you should endeavour to get your codebase into a state that you can turn the rule off ASAP.

**Comment by bbarry** [comment]
Btw `noImplicitAny` is the ts compiler option https://www.staging-typescript.org/tsconfig#noImplicitAny (only documentation link working as I type this comment) set in your tsconfig.
